src/lib/random_oracle/random_oracle.ml

```diff
@@ -106,6 +106,12 @@ let update ~state = update ~state params
 
 let hash ?init = hash ?init params
 
+let pow2 =
+  let rec pow2 acc n =
+    if n = 0 then acc else pow2 (Field.add acc acc) (n - 1)
+  in
+  Memo.general ~hashable:Int.hashable (fun n -> pow2 Field.one n)
```

Comment:
This is short for `Memoize`, it's from the [Jane Street `Core_kernel` library](https://ocaml.janestreet.com/ocaml-core/latest/doc/core_kernel/Core_kernel/Memo/index.html). It caches the result of a function so that subsequent calls don't recompute the value.